diff --git a/src/fft.hpp b/src/fft.hpp
index 1623525..274a43c 100644
--- a/src/fft.hpp
+++ b/src/fft.hpp
@@ -443,68 +443,46 @@ struct power_series : public std::vector<T> {
 		return r;
 	}
 
-	friend power_series stretch(const power_series& a, int n) {
-		power_series r(a.size());
-		for (int i = 0; i*n < int(a.size()); i++) {
-			r[i*n] = a[i];
-		}
-		return r;
-	}
-	friend power_series inverse(power_series a) {
-		power_series r(sz(a));
-		inverser::inverse(begin(a), sz(a), begin(r));
-		return r;
-	}
-	friend power_series deriv_shift(power_series a) {
-		for (int i = 0; i < a.len(); i++) {
-			a[i] *= i;
-		}
-		return a;
-	}
-	friend power_series integ_shift(power_series a) {
-		assert(a[0] == 0);
-		T f = 1;
-		for (int i = 1; i < int(a.size()); i++) {
-			a[i] *= f;
-			f *= i;
-		}
-		f = inv(f);
-		for (int i = int(a.size()) - 1; i > 0; i--) {
-			a[i] *= f;
-			f *= i;
-		}
-		return a;
-	}
-	friend power_series deriv_shift_log(power_series a) {
-		auto r = deriv_shift(a);
-		return r * inverse(a);
-	}
-	friend power_series poly_log(power_series a) {
-		assert(a[0] == 1);
-		return integ_shift(deriv_shift_log(std::move(a)));
-	}
-	friend power_series poly_exp(power_series a) {
-		assert(a.size() >= 1);
-		assert(a[0] == 0);
-		power_series r(1, T(1));
-		while (r.size() < a.size()) {
-			int n_sz = std::min(int(r.size()) * 2, int(a.size()));
-			r.resize(n_sz);
-			power_series v(a.begin(), a.begin() + n_sz);
-			v -= poly_log(r);
-			v[0] += 1;
-			r *= v;
-		}
-		return r;
-	}
-	friend power_series poly_pow_monic(power_series a, T k) {
-		if (a.empty()) return a;
-		assert(a.size() >= 1);
-		assert(a[0] == 1);
-		a = poly_log(a);
-		a *= k;
-		return poly_exp(a);
-	}
+        friend power_series stretch(const power_series& a, int n) {
+                power_series r(a.size());
+                for (int i = 0; i*n < int(a.size()); i++) {
+                        r[i*n] = a[i];
+                }
+                return r;
+        }
+        friend power_series inverse(const power_series& a) {
+                power_series r(sz(a));
+                inverser::inverse(begin(a), sz(a), begin(r));
+                return r;
+        }
+        friend power_series deriv_shift(const power_series& a) {
+                power_series r(a.len());
+                for (int i = 0; i < a.len(); i++) r[i] = a[i] * T(i);
+                return r;
+        }
+        friend power_series integ_shift(power_series a) {
+                assert(a[0] == 0);
+                T f = 1;
+                for (int i = 1; i < int(a.size()); i++) {
+                        a[i] *= f;
+                        f *= i;
+                }
+                f = inv(f);
+                for (int i = int(a.size()) - 1; i > 0; i--) {
+                        a[i] *= f;
+                        f *= i;
+                }
+                return a;
+        }
+        friend power_series deriv_shift_log(const power_series& a) {
+                power_series deriv = deriv_shift(a);
+                power_series ainv = inverse(a);
+                return deriv * ainv;
+        }
+        friend power_series poly_log(const power_series& a) {
+                assert(a[0] == 1);
+                return integ_shift(deriv_shift_log(a));
+        }
 	friend power_series poly_pow(power_series a, int64_t k) {
 		assert(k >= 0);
 		if (k == 0) {
