diff --git a/highs/mip/HighsMipSolver.cpp b/highs/mip/HighsMipSolver.cpp
index 8e3f9a067..c07ac5fb9 100644
--- a/highs/mip/HighsMipSolver.cpp
+++ b/highs/mip/HighsMipSolver.cpp
@@ -148,20 +148,7 @@ restart:
       mipdata_->queryExternalSolution(
           solution_objective_, kExternalMipSolutionQueryOriginAfterSetup);
 
-    if (options_mip_->mip_heuristic_run_feasibility_jump) {
-      // Apply the feasibility jump before evaluating the root node
-      analysis_.mipTimerStart(kMipClockFeasibilityJump);
-      HighsModelStatus returned_model_status = mipdata_->feasibilityJump();
-      analysis_.mipTimerStop(kMipClockFeasibilityJump);
-      if (modelstatus_ == HighsModelStatus::kNotset &&
-          returned_model_status == HighsModelStatus::kInfeasible) {
-        // feasibilityJump can spot trivial infeasibility, so act on it
-        modelstatus_ = returned_model_status;
-        cleanupSolve();
-        return;
-      }
-    }
-    // Apply the trivial heuristics
+    // Apply the trivial heuristics before the feasibility jump
     analysis_.mipTimerStart(kMipClockTrivialHeuristics);
     HighsModelStatus returned_model_status = mipdata_->trivialHeuristics();
     analysis_.mipTimerStop(kMipClockTrivialHeuristics);
@@ -172,10 +159,25 @@ restart:
       cleanupSolve();
       return;
     }
+
+    if (options_mip_->mip_heuristic_run_feasibility_jump) {
+      // Apply the feasibility jump after trivial heuristics
+      analysis_.mipTimerStart(kMipClockFeasibilityJump);
+      HighsModelStatus returned_model_status_fj = mipdata_->feasibilityJump();
+      analysis_.mipTimerStop(kMipClockFeasibilityJump);
+      if (modelstatus_ == HighsModelStatus::kNotset &&
+          returned_model_status_fj == HighsModelStatus::kInfeasible) {
+        // feasibilityJump can spot trivial infeasibility, so act on it
+        modelstatus_ = returned_model_status_fj;
+        cleanupSolve();
+        return;
+      }
+    }
+
     if (analysis_.analyse_mip_time && !submip)
       if (analysis_.analyse_mip_time & !submip)
         highsLogUser(options_mip_->log_options, HighsLogType::kInfo,
-                     "MIP-Timing: %11.2g - starting evaluate root node\n",
+                     "MIP-Timing: %11.2g - starting  evaluate root node\n",
                      timer_.read());
     analysis_.mipTimerStart(kMipClockEvaluateRootNode);
     mipdata_->evaluateRootNode();
