diff --git a/hwy/ops/generic_ops-inl.h b/hwy/ops/generic_ops-inl.h
index 01206b7a..7d2626ef 100644
--- a/hwy/ops/generic_ops-inl.h
+++ b/hwy/ops/generic_ops-inl.h
@@ -1269,9 +1269,11 @@ HWY_API Vec128<uint64_t> operator*(Vec128<uint64_t> x, Vec128<uint64_t> y) {
   return lolo + ShiftLeft<32>(lohi + hilo);
 }
 HWY_API Vec128<int64_t> operator*(Vec128<int64_t> x, Vec128<int64_t> y) {
-  return BitCast(Simd<int64_t,2,0>(), BitCast(Simd<uint64_t,2,0>(), x) * BitCast(Simd<uint64_t,2,0>(), y));
+  const auto ux = BitCast(Simd<uint64_t,2,0>(), x);
+  const auto uy = BitCast(Simd<uint64_t,2,0>(), y);
+  return BitCast(Simd<int64_t,2,0>(), ux * uy);
 }
-#if HWY_TARGET == HWY_AVX2
+#if HWY_TARGET == HWY_AVX2 || HWY_TARGET == HWY_HASWELL
 
 HWY_API Vec256<uint64_t> operator*(Vec256<uint64_t> x, Vec256<uint64_t> y) {
   Vec256<uint32_t> x32 = BitCast(Simd<uint32_t,8,0>(), x);
@@ -1282,9 +1284,12 @@ HWY_API Vec256<uint64_t> operator*(Vec256<uint64_t> x, Vec256<uint64_t> y) {
   return lolo + ShiftLeft<32>(lohi + hilo);
 }
 HWY_API Vec256<int64_t> operator*(Vec256<int64_t> x, Vec256<int64_t> y) {
-  return BitCast(Simd<int64_t,4,0>(), BitCast(Simd<uint64_t,4,0>(), x) * BitCast(Simd<uint64_t,4,0>(), y));
+  const auto ux = BitCast(Simd<uint64_t,4,0>(), x);
+  const auto uy = BitCast(Simd<uint64_t,4,0>(), y);
+  return BitCast(Simd<int64_t,4,0>(), ux * uy);
 }
 #endif 
+
 #endif // HWY_NATIVE_I64MULLO
 
 #endif  // HWY_NATIVE_POPCNT
