diff --git a/includes/acl/decompression/impl/steps/rotation_constant.h b/includes/acl/decompression/impl/steps/rotation_constant.h
index af3f4707..daf7ea26 100644
--- a/includes/acl/decompression/impl/steps/rotation_constant.h
+++ b/includes/acl/decompression/impl/steps/rotation_constant.h
@@ -132,6 +132,7 @@ namespace acl
 			// prefetching every entry, the ptr will remain forever null and this
 			// the branch is always constant: not zero for some time, then forever zero
 			const void* next_prefetch_ptr = *prefetch_queue_ptr;
+				bool prefetching_done = (next_prefetch_ptr == nullptr);
 
 			// Tuned to give us the right number of instructions per sub-step
 			// On ARM64, 4 yields 78-103 instructions (68-93 with vst4q_f32):
diff --git a/includes/acl/decompression/impl/steps/rotation_default.h b/includes/acl/decompression/impl/steps/rotation_default.h
index 3822b9a0..25b7412d 100644
--- a/includes/acl/decompression/impl/steps/rotation_default.h
+++ b/includes/acl/decompression/impl/steps/rotation_default.h
@@ -63,6 +63,7 @@ namespace acl
 			// prefetching every entry, the ptr will remain forever null and this
 			// the branch is always constant: not zero for some time, then forever zero
 			const void* next_prefetch_ptr = *prefetch_queue_ptr;
+				bool prefetching_done = (next_prefetch_ptr == nullptr);
 
 			// Grab our constant default rotation if we have one, otherwise init with some value
 			const rtm::quatf default_rotation = default_mode == default_sub_track_mode::constant ? writer.get_constant_default_rotation() : rtm::quat_identity();
diff --git a/includes/acl/decompression/impl/steps/scale_default.h b/includes/acl/decompression/impl/steps/scale_default.h
index e6b512af..29896eb2 100644
--- a/includes/acl/decompression/impl/steps/scale_default.h
+++ b/includes/acl/decompression/impl/steps/scale_default.h
@@ -62,6 +62,8 @@ namespace acl
 			// prefetching every entry, the ptr will remain forever null and this
 			// the branch is always constant: not zero for some time, then forever zero
 			const void* next_prefetch_ptr = *prefetch_queue_ptr;
+			bool prefetching_done = (next_prefetch_ptr == nullptr);
+
 
 			// Grab our constant default scale if we have one, otherwise init with some value
 			rtm::vector4f default_scale;
