diff --git a/includes/acl/decompression/impl/steps/animated_unpack_vec3.h b/includes/acl/decompression/impl/steps/animated_unpack_vec3.h
index 1d0fa653..fc67d545 100644
--- a/includes/acl/decompression/impl/steps/animated_unpack_vec3.h
+++ b/includes/acl/decompression/impl/steps/animated_unpack_vec3.h
@@ -70,6 +70,9 @@ namespace acl
 			uint32_t animated_track_data_bit_offset = segment_sampling_context.animated_track_data_bit_offset;
 
 			const uint8_t* clip_range_data = clip_sampling_context.clip_range_data;
+			const uint32_t clip_range_entry_size = 3 * sizeof(float);
+			const uint8_t* clip_range_ptr = clip_range_data;
+			
 
 			rtm::vector4f* output_scratch_ptr = &output_scratch[0];
 
