diff --git a/src/quick-lint-js/buffering-visitor.h b/src/quick-lint-js/buffering-visitor.h
index 720fe526..321a2edf 100644
--- a/src/quick-lint-js/buffering-visitor.h
+++ b/src/quick-lint-js/buffering-visitor.h
@@ -5,6 +5,7 @@
 #define QUICK_LINT_JS_BUFFERING_VISITOR_H
 
 #include <boost/container/pmr/memory_resource.hpp>
+#include <boost/container/pmr/polymorphic_allocator.hpp>
 #include <optional>
 #include <quick-lint-js/language.h>
 #include <quick-lint-js/lex.h>
@@ -12,6 +13,7 @@
 #include <quick-lint-js/vector.h>
 #include <quick-lint-js/warning.h>
 #include <utility>
+#include <deque>
 #include <vector>
 
 QLJS_WARNING_PUSH
@@ -21,7 +23,7 @@ namespace quick_lint_js {
 class buffering_visitor {
  public:
   explicit buffering_visitor(boost::container::pmr::memory_resource *memory)
-      : visits_("buffering_visitor", vector<visit>::allocator_type(memory)) {}
+      : visits_(std::deque<visit, boost::container::pmr::polymorphic_allocator<visit>>::allocator_type(memory)) {}
 
   template <QLJS_PARSE_VISITOR Visitor>
   void move_into(Visitor &target) {
@@ -253,7 +255,7 @@ class buffering_visitor {
     };
   };
 
-  vector<visit> visits_;
+  std::deque<visit, boost::container::pmr::polymorphic_allocator<visit>> visits_;
 };
 QLJS_STATIC_ASSERT_IS_PARSE_VISITOR(buffering_visitor);
 }
