diff --git a/src/quick-lint-js/container/vector-profiler-debug.cpp b/src/quick-lint-js/container/vector-profiler-debug.cpp
index 5b9bd36d..15e67d86 100644
--- a/src/quick-lint-js/container/vector-profiler-debug.cpp
+++ b/src/quick-lint-js/container/vector-profiler-debug.cpp
@@ -76,17 +76,22 @@ vector_instrumentation::max_size_histogram_by_owner() const {
   std::map<std::string_view, std::map<std::size_t, int>> histogram;
   hash_map<std::pair<const char *, std::uintptr_t>, std::size_t> object_sizes;
   for (const vector_instrumentation::entry &entry : this->entries_) {
-    std::pair key(entry.owner, entry.object_id);
-    std::size_t &object_size = object_sizes[key];
-    object_size = std::max(entry.size, object_size);
+    std::pair<const char *, std::uintptr_t> key(entry.owner, entry.object_id);
+    auto it = object_sizes.find(key);
+    if (it == object_sizes.end()) {
+      auto emplaced = object_sizes.emplace(key, entry.size);
+      it = emplaced.first;
+    } else if (entry.size > it->second) {
+      it->second = entry.size;
+    }
 
     if (entry.event == event::destroy) {
-      histogram[entry.owner][object_size] += 1;
-      object_sizes.erase(key);
+      histogram[entry.owner][it->second] += 1;
+      object_sizes.erase(it);
     }
   }
-  for (auto &[owner_and_object_id, size] : object_sizes) {
-    histogram[owner_and_object_id.first][size] += 1;
+  for (auto &kv : object_sizes) {
+    histogram[kv.first.first][kv.second] += 1;
   }
   return histogram;
 }
