diff --git a/src/parse-expression.cpp b/src/parse-expression.cpp
index dbaa7d27..1d80efc2 100644
--- a/src/parse-expression.cpp
+++ b/src/parse-expression.cpp
@@ -464,7 +464,7 @@ expression* parser::parse_primary_expression(parse_visitor_base& v,
   // async           // Identifier.
   // async () => {}  // Arrow function.
   case token_type::kw_async: {
-    token async_token = this->peek();
+    const token &async_token = this->peek();
     this->skip();
     return this->parse_async_expression(v, async_token, prec);
   }
@@ -686,7 +686,7 @@ expression* parser::parse_primary_expression(parse_visitor_base& v,
 }
 
 expression* parser::parse_async_expression(parse_visitor_base& v,
-                                           token async_token, precedence prec) {
+                                           const token &async_token, precedence prec) {
   expression* ast = this->parse_async_expression_only(
       v, async_token, /*allow_in_operator=*/prec.in_operator);
   if (!prec.binary_operators) {
@@ -696,7 +696,7 @@ expression* parser::parse_async_expression(parse_visitor_base& v,
 }
 
 expression* parser::parse_async_expression_only(parse_visitor_base& v,
-                                                token async_token,
+                                                const token &async_token,
                                                 bool allow_in_operator) {
   const char8* async_begin = async_token.begin;
 
@@ -846,7 +846,7 @@ expression* parser::parse_async_expression_only(parse_visitor_base& v,
 }
 
 expression* parser::parse_await_expression(parse_visitor_base& v,
-                                           token await_token, precedence prec) {
+                                           const token &await_token, precedence prec) {
   bool is_identifier = [&]() -> bool {
     if (this->in_async_function_ ||
         (this->in_top_level_ &&
diff --git a/src/parse-statement.cpp b/src/parse-statement.cpp
index cd01e8a6..e0e4c52b 100644
--- a/src/parse-statement.cpp
+++ b/src/parse-statement.cpp
@@ -2704,7 +2704,7 @@ void parser::parse_and_visit_variable_declaration_statement(
 }
 
 void parser::parse_and_visit_let_bindings(parse_visitor_base &v,
-                                          token declaring_token,
+                                          const token &declaring_token,
                                           bool allow_in_operator,
                                           bool allow_const_without_initializer,
                                           bool is_in_for_initializer) {
diff --git a/src/parse.cpp b/src/parse.cpp
index 8a391e48..db342253 100644
--- a/src/parse.cpp
+++ b/src/parse.cpp
@@ -441,7 +441,7 @@ parser::try_parse_function_with_leading_star() {
 }
 
 bool parser::is_let_token_a_variable_reference(
-    token following_token, bool allow_declarations) noexcept {
+    const token &following_token, bool allow_declarations) noexcept {
   switch (following_token.type) {
   QLJS_CASE_BINARY_ONLY_OPERATOR_SYMBOL:
   QLJS_CASE_COMPOUND_ASSIGNMENT_OPERATOR:
diff --git a/src/quick-lint-js/parse.h b/src/quick-lint-js/parse.h
index 4e903344..4b8623cf 100644
--- a/src/quick-lint-js/parse.h
+++ b/src/quick-lint-js/parse.h
@@ -225,17 +225,17 @@ class parser {
 
   void parse_and_visit_variable_declaration_statement(parse_visitor_base &v);
   void parse_and_visit_let_bindings(
-      parse_visitor_base &v, token declaring_token, bool allow_in_operator,
+      parse_visitor_base &v, const token &declaring_token, bool allow_in_operator,
       bool allow_const_without_initializer = false,
       bool is_in_for_initializer = false);
   // declaring_token is the const/let/var token.
   void parse_and_visit_let_bindings(parse_visitor_base &v,
-                                    token declaring_token,
+                                    const token &declaring_token,
                                     variable_kind declaration_kind,
                                     bool allow_in_operator,
                                     bool allow_const_without_initializer,
                                     bool is_in_for_initializer);
-  bool is_let_token_a_variable_reference(token following_token,
+  bool is_let_token_a_variable_reference(const token &following_token,
                                          bool allow_declarations) noexcept;
   void visit_binding_element(expression *ast, parse_visitor_base &v,
                              variable_kind declaration_kind,
@@ -308,12 +308,12 @@ class parser {
 
   expression *parse_expression(parse_visitor_base &, precedence);
   expression *parse_primary_expression(parse_visitor_base &, precedence);
-  expression *parse_async_expression(parse_visitor_base &, token async_token,
+  expression *parse_async_expression(parse_visitor_base &, const token &async_token,
                                      precedence);
   expression *parse_async_expression_only(parse_visitor_base &,
-                                          token async_token,
+                                          const token &async_token,
                                           bool allow_in_operator);
-  expression *parse_await_expression(parse_visitor_base &, token await_token,
+  expression *parse_await_expression(parse_visitor_base &, const token &await_token,
                                      precedence prec);
   expression *parse_expression_remainder(parse_visitor_base &, expression *,
                                          precedence);
