diff --git a/src/arg-parser.cpp b/src/arg-parser.cpp
index 1ccedc38..848e94b1 100644
--- a/src/arg-parser.cpp
+++ b/src/arg-parser.cpp
@@ -35,7 +35,9 @@ bool arg_parser::match_flag_shorthand(char option_shorthand) noexcept {
   if (!this->option_.has_value()) {
     return false;
   }
-  bool matches = this->option_->arg_key == std::string{'-', option_shorthand};
+  bool matches = this->option_->arg_key.size() == 2 &&
+                 this->option_->arg_key[0] == '-' &&
+                 this->option_->arg_key[1] == option_shorthand;
 
   if (matches) {
     this->advance(1);
