diff --git a/src/quick-lint-js/event-loop.h b/src/quick-lint-js/event-loop.h
index fa56b9bf..38f6645b 100644
--- a/src/quick-lint-js/event-loop.h
+++ b/src/quick-lint-js/event-loop.h
@@ -9,6 +9,7 @@
 #else
 
 #include <array>
+#include <vector>
 #include <cstddef>
 #include <cstdint>
 #include <optional>
@@ -90,8 +91,9 @@ class event_loop_base {
   // Returns true when the pipe has closed. Returns false if the pipe might
   // still have data available (now or in the future).
   bool read_from_pipe() {
+    // Increased buffer size to reduce number of syscalls when reading from pipe.
     // TODO(strager): Pick buffer size intelligently.
-    std::array<char8, 1024> buffer;
+    std::array<char8, 65536> buffer;
     platform_file_ref pipe = this->const_derived().get_readable_pipe();
 #if QLJS_EVENT_LOOP_READ_PIPE_NON_BLOCKING
     QLJS_SLOW_ASSERT(pipe.is_pipe_non_blocking());
