diff --git a/include/sqlpp11/sqlite3/prepared_statement.h b/include/sqlpp11/sqlite3/prepared_statement.h
index 89220a0f..4c84232c 100644
--- a/include/sqlpp11/sqlite3/prepared_statement.h
+++ b/include/sqlpp11/sqlite3/prepared_statement.h
@@ -256,7 +256,7 @@ namespace sqlpp
         int result;
         if (not is_null)
         {
-          const auto time = ::date::make_time(::sqlpp::chrono::floor<::std::chrono::microseconds>(*value));
+          const auto time = ::date::make_time(*value);
           std::ostringstream os;  // gcc-4.9 does not support auto os = std::ostringstream{};
           os << time;
           const auto text = os.str();
