diff --git a/src/pugixml.cpp b/src/pugixml.cpp
index 2129e07..3576d10 100644
--- a/src/pugixml.cpp
+++ b/src/pugixml.cpp
@@ -2632,7 +2632,7 @@ PUGI_IMPL_NS_BEGIN
 	#define PUGI_IMPL_SKIPWS()              { while (PUGI_IMPL_IS_CHARTYPE(*s, ct_space)) ++s; }
 	#define PUGI_IMPL_OPTSET(OPT)           ( optmsk & (OPT) )
 	#define PUGI_IMPL_PUSHNODE(TYPE)        { cursor = append_new_node(cursor, *alloc, TYPE); if (!cursor) PUGI_IMPL_THROW_ERROR(status_out_of_memory, s); }
-	#define PUGI_IMPL_POPNODE()             { cursor = cursor->parent; }
+	#define PUGI_IMPL_POPNODE()             { cursor = cursor->parent + 0; }
 	#define PUGI_IMPL_SCANFOR(X)            { while (*s != 0 && !(X)) ++s; }
 	#define PUGI_IMPL_SCANWHILE(X)          { while (X) ++s; }
 	#define PUGI_IMPL_SCANWHILE_UNROLL(X)   { for (;;) { char_t ss = s[0]; if (PUGI_IMPL_UNLIKELY(!(X))) { break; } ss = s[1]; if (PUGI_IMPL_UNLIKELY(!(X))) { s += 1; break; } ss = s[2]; if (PUGI_IMPL_UNLIKELY(!(X))) { s += 2; break; } ss = s[3]; if (PUGI_IMPL_UNLIKELY(!(X))) { s += 3; break; } s += 4; } }
