diff --git a/include/spdlog/logger.h b/include/spdlog/logger.h
index 05a40a52..ddfd2ff6 100644
--- a/include/spdlog/logger.h
+++ b/include/spdlog/logger.h
@@ -364,7 +364,7 @@ protected:
         SPDLOG_TRY
         {
             memory_buf_t buf;
-            fmt_lib::vformat_to(std::back_inserter(buf), fmt, fmt_lib::make_format_args(std::forward<Args>(args)...));
+            fmt_lib::detail::vformat_to(buf, fmt, fmt_lib::make_format_args(std::forward<Args>(args)...));
 
             details::log_msg log_msg(loc, name_, lvl, string_view_t(buf.data(), buf.size()));
             log_it_(log_msg, log_enabled, traceback_enabled);
@@ -386,7 +386,7 @@ protected:
         {
             // format to wmemory_buffer and convert to utf8
             wmemory_buf_t wbuf;
-            fmt_lib::vformat_to(std::back_inserter(wbuf), fmt, fmt_lib::make_format_args<fmt_lib::wformat_context>(std::forward<Args>(args)...));
+            fmt_lib::detail::vformat_to(wbuf, fmt, fmt_lib::make_format_args<fmt_lib::wformat_context>(std::forward<Args>(args)...));
 
             memory_buf_t buf;
             details::os::wstr_to_utf8buf(wstring_view_t(wbuf.data(), wbuf.size()), buf);
