diff --git a/hwy/contrib/sort/traits128-inl.h b/hwy/contrib/sort/traits128-inl.h
index 7f7d69f7..676cc300 100644
--- a/hwy/contrib/sort/traits128-inl.h
+++ b/hwy/contrib/sort/traits128-inl.h
@@ -60,7 +60,7 @@ struct Traits128 : public Order {
   // For HeapSort
   HWY_INLINE void Swap(uint64_t* a, uint64_t* b) const {
     for (size_t i = 0; i < LanesPerKey(); ++i) {
-      const T temp = a[i];
+      const LaneType temp = a[i];
       a[i] = b[i];
       b[i] = temp;
     }
diff --git a/hwy/ops/arm_sve-inl.h b/hwy/ops/arm_sve-inl.h
index c6e332b5..f05e8f82 100644
--- a/hwy/ops/arm_sve-inl.h
+++ b/hwy/ops/arm_sve-inl.h
@@ -1460,11 +1460,11 @@ HWY_API VFromD<D> Iota(const D d, TFromD<D> first) {
 namespace detail {
 
 template <class D>
-svbool_t MaskLowerHalf(D d) {
+HWY_INLINE svbool_t MaskLowerHalf(D d) {
   return FirstN(d, Lanes(d) / 2);
 }
 template <class D>
-svbool_t MaskUpperHalf(D d) {
+HWY_INLINE svbool_t MaskUpperHalf(D d) {
   // For Splice to work as intended, make sure bits above Lanes(d) are zero.
   return AndNot(MaskLowerHalf(d), detail::MakeMask(d));
 }
