diff --git a/snappy.cc b/snappy.cc
index 421b335..1d7c72f 100644
--- a/snappy.cc
+++ b/snappy.cc
@@ -434,11 +434,7 @@ static inline bool Copy64BytesWithPatternExtension(char* dst, size_t offset) {
     // The simpler formulation of `dst[i - offset]` induces undefined behavior.
     for (int i = 0; i < 16; i++) dst[i] = (dst - offset)[i];
     // Find a multiple of pattern >= 16.
-    static std::array<uint8_t, 16> pattern_sizes = []() {
-      std::array<uint8_t, 16> res;
-      for (int i = 1; i < 16; i++) res[i] = (16 / i + 1) * i;
-      return res;
-    }();
+    constexpr std::array<uint8_t, 16> pattern_sizes = {0, 17, 18, 18, 20, 20, 18, 21, 24, 18, 20, 22, 24, 26, 28, 30};
     offset = pattern_sizes[offset];
     for (int i = 1; i < 4; i++) {
       std::memcpy(dst + i * 16, dst + i * 16 - offset, 16);
diff --git a/third_party/benchmark b/third_party/benchmark
--- a/third_party/benchmark
+++ b/third_party/benchmark
@@ -1 +1 @@
-Subproject commit d572f4777349d43653b21d6c2fc63020ab326db2
+Subproject commit d572f4777349d43653b21d6c2fc63020ab326db2-dirty
diff --git a/third_party/googletest b/third_party/googletest
--- a/third_party/googletest
+++ b/third_party/googletest
@@ -1 +1 @@
-Subproject commit b796f7d44681514f58a683a3a71ff17c94edb0c1
+Subproject commit b796f7d44681514f58a683a3a71ff17c94edb0c1-dirty
