diff --git a/snappy.cc b/snappy.cc
index 3f446c6..d9eeb4e 100644
--- a/snappy.cc
+++ b/snappy.cc
@@ -1340,6 +1340,9 @@ class SnappyDecompressor {
           const ptrdiff_t entry = kLengthMinusOffset[c];
           preload = LittleEndian::Load32(ip);
           const uint32_t trailer = ExtractLowBytes(preload, c & 3);
+#if defined(__GNUC__) && defined(__x86_64__)
+          asm("") : : "r"(trailer);
+#endif
           const uint32_t length = entry & 0xff;
           assert(length > 0);
 
