diff --git a/external/benchmark b/external/benchmark
--- a/external/benchmark
+++ b/external/benchmark
@@ -1 +1 @@
-Subproject commit 0d98dba29d66e93259db7daa53a9327df767a415
+Subproject commit 0d98dba29d66e93259db7daa53a9327df767a415-dirty
diff --git a/external/catch2 b/external/catch2
--- a/external/catch2
+++ b/external/catch2
@@ -1 +1 @@
-Subproject commit c4e3767e265808590986d5db6ca1b5532a7f3d13
+Subproject commit c4e3767e265808590986d5db6ca1b5532a7f3d13-dirty
diff --git a/external/rtm b/external/rtm
--- a/external/rtm
+++ b/external/rtm
@@ -1 +1 @@
-Subproject commit 2a8570ec3cbb5e059861e732377c255be9181863
+Subproject commit 2a8570ec3cbb5e059861e732377c255be9181863-dirty
diff --git a/external/sjson-cpp b/external/sjson-cpp
--- a/external/sjson-cpp
+++ b/external/sjson-cpp
@@ -1 +1 @@
-Subproject commit 294331edbc68a0709c167b2369adbcf3eb430daa
+Subproject commit 294331edbc68a0709c167b2369adbcf3eb430daa-dirty
diff --git a/includes/acl/compression/impl/compact_constant_streams.h b/includes/acl/compression/impl/compact_constant_streams.h
index 763627ff..b904abec 100644
--- a/includes/acl/compression/impl/compact_constant_streams.h
+++ b/includes/acl/compression/impl/compact_constant_streams.h
@@ -84,8 +84,8 @@ namespace acl
 			const uint32_t dirty_transform_indices[2] = { 0, 1 };
 			rtm::qvvf local_transforms[2];
 			rtm::qvvf base_transforms[2];
-			uint8_t local_transforms_converted[1024];	// Big enough for 2 transforms for sure
-			uint8_t base_transforms_converted[1024];	// Big enough for 2 transforms for sure
+			alignas(16) uint8_t local_transforms_converted[1024];	// Big enough for 2 transforms for sure
+			alignas(16) uint8_t base_transforms_converted[1024];	// Big enough for 2 transforms for sure
 
 			const size_t metric_transform_size = error_metric.get_transform_size(lossy_clip_context.has_scale);
 			ACL_ASSERT(metric_transform_size * 2 <= sizeof(local_transforms_converted), "Transform size is too large");
diff --git a/includes/acl/compression/impl/optimize_looping.h b/includes/acl/compression/impl/optimize_looping.h
index fb496ac0..6fe8a0a8 100644
--- a/includes/acl/compression/impl/optimize_looping.h
+++ b/includes/acl/compression/impl/optimize_looping.h
@@ -84,8 +84,8 @@ namespace acl
 			const uint32_t dirty_transform_indices[2] = { 0, 1 };
 			rtm::qvvf local_transforms[2];
 			rtm::qvvf base_transforms[2];
-			uint8_t local_transforms_converted[1024];	// Big enough for 2 transforms for sure
-			uint8_t base_transforms_converted[1024];	// Big enough for 2 transforms for sure
+			alignas(16) uint8_t local_transforms_converted[1024];	// Big enough for 2 transforms for sure
+			alignas(16) uint8_t base_transforms_converted[1024];	// Big enough for 2 transforms for sure
 
 			const size_t metric_transform_size = error_metric.get_transform_size(context.has_scale);
 			ACL_ASSERT(metric_transform_size * 2 <= sizeof(local_transforms_converted), "Transform size is too large");
