diff --git a/includes/acl/compression/transform_error_metrics.h b/includes/acl/compression/transform_error_metrics.h
index c47a64b2..bdb43e4f 100644
--- a/includes/acl/compression/transform_error_metrics.h
+++ b/includes/acl/compression/transform_error_metrics.h
@@ -38,7 +38,7 @@
 #include <rtm/qvvf.h>
 #include <rtm/scalarf.h>
 
-//#define ACL_IMPL_USE_SOA_ERROR_METRIC
+#define ACL_IMPL_USE_SOA_ERROR_METRIC
 
 ACL_IMPL_FILE_PRAGMA_PUSH
 
