diff --git a/src/oatpp/core/data/buffer/Processor.cpp b/src/oatpp/core/data/buffer/Processor.cpp
index b5db589a..d7c8656a 100644
--- a/src/oatpp/core/data/buffer/Processor.cpp
+++ b/src/oatpp/core/data/buffer/Processor.cpp
@@ -88,9 +88,8 @@ void InlineWriteData::setEof() {
 ProcessingPipeline::ProcessingPipeline(const std::vector<base::ObjectHandle<Processor>>& processors)
   : m_processors(processors)
 {
-  for(v_int32 i = 0; i < (v_int32) m_processors.size() - 1; i ++) {
-    m_intermediateData.push_back(data::buffer::InlineReadData());
-  }
+  const std::size_t count = m_processors.size() > 0 ? m_processors.size() - 1 : 0;
+  m_intermediateData = std::vector<data::buffer::InlineReadData>(count);
 }
 
 v_io_size ProcessingPipeline::suggestInputStreamReadSize() {
