diff --git a/src/quick-lint-js/parse.h b/src/quick-lint-js/parse.h
index a072d768..fd72fc14 100644
--- a/src/quick-lint-js/parse.h
+++ b/src/quick-lint-js/parse.h
@@ -243,14 +243,14 @@ class parser {
                              variable_init_kind init_kind);
 
   struct precedence {
-    bool binary_operators = true;
-    bool math_or_logical_or_assignment = true;
-    bool commas = true;
-    bool in_operator = true;
+    unsigned binary_operators : 1 = 1;
+    unsigned math_or_logical_or_assignment : 1 = 1;
+    unsigned commas : 1 = 1;
+    unsigned in_operator : 1 = 1;
 
     // If true, parse unexpected trailing identifiers as part of the
     // expression (and emit an error).
-    bool trailing_identifiers = false;
+    unsigned trailing_identifiers : 1 = 0;
 
     // If true, try parsing a trailing '{' as the body of an arrow function. For
     // example:
@@ -259,9 +259,9 @@ class parser {
     //    ^ missing '=>'
     //
     // If false, stop parsing at a trailing '{' and do not report an error.
-    bool trailing_curly_is_arrow_body = true;
+    unsigned trailing_curly_is_arrow_body : 1 = 1;
 
-    bool conditional_operator = true;
+    unsigned conditional_operator : 1 = 1;
   };
 
   // binary_expression_builder helps in the creation of a
