diff --git a/src/output-stream.cpp b/src/output-stream.cpp
index 704fac64..0f9b73fe 100644
--- a/src/output-stream.cpp
+++ b/src/output-stream.cpp
@@ -21,7 +21,9 @@ output_stream::output_stream()
 
 output_stream::output_stream(int buffer_size)
     : buffer_(new char8[narrow_cast<std::size_t>(buffer_size)]),
-      buffer_size_(buffer_size) {}
+      buffer_size_(buffer_size) {
+  this->cursor_ = this->buffer_.get();
+}
 
 output_stream::~output_stream() = default;
 
@@ -46,18 +48,18 @@ QLJS_WARNING_POP
 
 void output_stream::flush() {
   this->flush_impl(string8_view(this->buffer_.get(),
-                                narrow_cast<std::size_t>(this->cursor_)));
-  this->cursor_ = 0;
+                                narrow_cast<std::size_t>(this->cursor_ - this->buffer_.get())));
+  this->cursor_ = this->buffer_.get();
 }
 
 char8* output_stream::reserve(int byte_count) {
   if (byte_count > this->buffer_size_) {
     return nullptr;
   }
-  if (this->cursor_ + byte_count >= this->buffer_size_) {
+  if ((this->cursor_ - this->buffer_.get()) + byte_count >= this->buffer_size_) {
     this->flush();
   }
-  char8* out = this->buffer_.get() + this->cursor_;
+  char8* out = this->cursor_;
   this->cursor_ += byte_count;
   return out;
 }
diff --git a/src/quick-lint-js/output-stream.h b/src/quick-lint-js/output-stream.h
index 2fc6fc0f..b858a019 100644
--- a/src/quick-lint-js/output-stream.h
+++ b/src/quick-lint-js/output-stream.h
@@ -30,7 +30,7 @@ class output_stream {
     if (data) {
       int bytes_written = f(data);
       QLJS_ASSERT(bytes_written <= max_byte_count);
-      this->cursor_ = this->cursor_ - max_byte_count + bytes_written;
+      this->cursor_ = this->cursor_ + (bytes_written - max_byte_count);
     } else {
       // this->buffer_ is too small.
       this->flush();
@@ -68,7 +68,7 @@ class output_stream {
 
   std::unique_ptr<char8[]> buffer_;
   int buffer_size_;
-  int cursor_ = 0;
+  char8* cursor_ = nullptr;
 };
 
 #if defined(__EMSCRIPTEN__)
