diff --git a/src/quick-lint-js/container/vector.h b/src/quick-lint-js/container/vector.h
index a2b1701b..02d760e4 100644
--- a/src/quick-lint-js/container/vector.h
+++ b/src/quick-lint-js/container/vector.h
@@ -12,6 +12,13 @@
 #include <quick-lint-js/container/winkable.h>
 #include <quick-lint-js/feature.h>
 #include <quick-lint-js/port/attribute.h>
+#if defined(__clang__) || defined(__GNUC__)
+#define QLJS_NOINLINE __attribute__((noinline))
+#elif defined(_MSC_VER)
+#define QLJS_NOINLINE __declspec(noinline)
+#else
+#define QLJS_NOINLINE /* empty */
+#endif
 #include <quick-lint-js/port/span.h>
 #include <quick-lint-js/util/cast.h>
 #include <string_view>
@@ -358,7 +365,7 @@ class Raw_Bump_Vector {
 
  private:
   // Growth strategy.
-  void reserve_grow_by_at_least(size_type minimum_new_entries) {
+  QLJS_NOINLINE void reserve_grow_by_at_least(size_type minimum_new_entries) {
     size_type old_capacity = this->capacity();
     constexpr size_type minimum_capacity = 4;
     size_type new_size = (std::max)(
