diff --git a/src/quick-lint-js/fe/identifier.h b/src/quick-lint-js/fe/identifier.h
index 48312f3b..a7927823 100644
--- a/src/quick-lint-js/fe/identifier.h
+++ b/src/quick-lint-js/fe/identifier.h
@@ -7,7 +7,6 @@
 #include <cstddef>
 #include <quick-lint-js/fe/source-code-span.h>
 #include <quick-lint-js/port/char8.h>
-#include <quick-lint-js/util/narrow-cast.h>
 
 namespace quick_lint_js {
 class identifier {
@@ -16,14 +15,14 @@ class identifier {
   explicit identifier(source_code_span span) noexcept
       : span_begin_(span.begin()),
         normalized_begin_(this->span_begin_),
-        span_size_(narrow_cast<int>(span.end() - span.begin())),
+        span_size_(static_cast<int>(span.end() - span.begin())),
         normalized_size_(this->span_size_) {}
 
   explicit identifier(source_code_span span, string8_view normalized) noexcept
       : span_begin_(span.begin()),
         normalized_begin_(normalized.data()),
-        span_size_(narrow_cast<int>(span.end() - span.begin())),
-        normalized_size_(narrow_cast<int>(normalized.size())) {}
+        span_size_(static_cast<int>(span.end() - span.begin())),
+        normalized_size_(static_cast<int>(normalized.size())) {}
 
   explicit identifier(source_code_span span,
                       const char8* normalized) noexcept = delete;
