diff --git a/src/quick-lint-js/container/linked-bump-allocator.h b/src/quick-lint-js/container/linked-bump-allocator.h
index 4bb1a792..a1964e25 100644
--- a/src/quick-lint-js/container/linked-bump-allocator.h
+++ b/src/quick-lint-js/container/linked-bump-allocator.h
@@ -178,9 +178,9 @@ class Linked_Bump_Allocator final : public Memory_Resource {
   [[nodiscard]] Disable_Guard disable() { return Disable_Guard(this); }
 
  protected:
-  void* do_allocate(std::size_t bytes, std::size_t align) override;
+  void* do_allocate(std::size_t bytes, std::size_t align) noexcept override;
   void do_deallocate(void* p, std::size_t bytes,
-                     [[maybe_unused]] std::size_t align) override;
+                     [[maybe_unused]] std::size_t align) noexcept override;
 
  private:
   struct alignas(alignof(void*)) Chunk_Header {
diff --git a/src/quick-lint-js/port/memory-resource.h b/src/quick-lint-js/port/memory-resource.h
index 5d0e28cb..b135b0f6 100644
--- a/src/quick-lint-js/port/memory-resource.h
+++ b/src/quick-lint-js/port/memory-resource.h
@@ -11,18 +11,18 @@ class Memory_Resource {
  public:
   virtual ~Memory_Resource() = default;
 
-  void* allocate(std::size_t bytes, std::size_t alignment) {
+  void* allocate(std::size_t bytes, std::size_t alignment) noexcept {
     return this->do_allocate(bytes, alignment);
   }
 
-  void deallocate(void* p, std::size_t bytes, std::size_t alignment) {
+  void deallocate(void* p, std::size_t bytes, std::size_t alignment) noexcept {
     return this->do_deallocate(p, bytes, alignment);
   }
 
  protected:
-  virtual void* do_allocate(std::size_t bytes, std::size_t alignment) = 0;
+  virtual void* do_allocate(std::size_t bytes, std::size_t alignment) noexcept = 0;
   virtual void do_deallocate(void* p, std::size_t bytes,
-                             std::size_t alignment) = 0;
+                             std::size_t alignment) noexcept = 0;
 };
 
 // Like std::pmr::new_delete_resource.
