diff --git a/include/xsimd/arch/generic/xsimd_generic_details.hpp b/include/xsimd/arch/generic/xsimd_generic_details.hpp
index bc1b1cc..2a6659d 100644
--- a/include/xsimd/arch/generic/xsimd_generic_details.hpp
+++ b/include/xsimd/arch/generic/xsimd_generic_details.hpp
@@ -136,7 +136,7 @@ namespace xsimd {
 
       template <class A, class From, class To>
       struct conversion_type_impl<A, From, To,
-                void_t<decltype(fast_cast(std::declval<const From&>(), std::declval<const To&>(), std::declval<const A&>()))>>
+                void_t<decltype(fast_cast(std::declval<const batch<From, A>&>(), std::declval<const batch<To, A>&>(), std::declval<const A&>()))>>
       {
           using type = with_fast_conversion;
       };
