diff --git a/core/src/Pattern.h b/core/src/Pattern.h
index 66dd0031..d5699336 100644
--- a/core/src/Pattern.h
+++ b/core/src/Pattern.h
@@ -323,10 +323,10 @@ void GetPatternRow(Range<I> b_row, PatternRow& p_row)
 	if (*bitPos)
 		intPos++; // first value is number of white pixels, here 0
 
-#if defined(__AVX__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
+#if (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
 	// The following code as been observed to cause a speedup of up to 30% on large images on an AVX cpu
 	// but also to cause a 50% slowdown on an a Google Pixel 3 Android phone.
-	if constexpr (std::is_pointer_v<I> && sizeof(std::remove_pointer_t<I>) == 1) {
+	if constexpr (std::is_pointer_v<I> && sizeof(std::remove_pointer_t<I>) == 1 && sizeof(void*) == 8) {
 		using simd_t = uint64_t;
 		while (bitPos < bitPosEnd - sizeof(simd_t)) {
 			auto asSimd0 = BitHacks::LoadU<simd_t>(bitPos);
